### PR TITLE
Render notes with long and short form

### PR DESCRIPTION
**Pull request by coreycb**

### CODE REVIEW

**Comment by coreycb** [comment]
Tested with:
python ./scripts/render-notes.py -r "coreycb/thunderbird-notes" -p changelog -- ${APPLICATION_ID} ${FULL_VERSION_NAME} ${VERSION_CODE} ${GITHUB_NOTES_FILE} | tee -a $GITHUB_STEP_SUMMARY

**Comment by coreycb** [comment]
@kewisch This is ready for review.

File: .github/workflows/shippable_builds.yml [hunk 1]
```diff
@@ -204,17 +205,25 @@ jobs:
...
           mkdir -p ./app-metadata/${APPLICATION_ID}/en-US/changelogs
-          python ./scripts/render-notes.py ${APPLICATION_ID} ${FULL_VERSION_NAME} ${VERSION_CODE} | tee -a $GITHUB_STEP_SUMMARY
+          GITHUB_NOTES_FILE="$(mktemp -d)/long-notes.txt"
+          python ./scripts/render-notes.py -p changelog -- ${APPLICATION_ID} ${FULL_VERSION_NAME} ${VERSION_CODE} ${GITHUB_NOTES_FILE} | tee -a $GITHUB_STEP_SUMMARY
           echo "</pre>" | tee -a $GITHUB_STEP_SUMMARY
+          if [[ "$APP_NAME" == "thunderbird" ]]; then
+            echo "thunderbird_github_notes<<EOF" >> $GITHUB_OUTPUT
+            cat $GITHUB_NOTES_FILE >> $GITHUB_OUTPUT
+            echo "EOF" >> $GITHUB_OUTPUT
+          fi
```
Comment (kewisch) [inline .github/workflows/shippable_builds.yml:226]: Both K-9 Mail and Thunderbird for Android have GitHub Releases, so they both have long form notes. There are a few items that will be in the Thunderbird release notes that won't be in the K-9 Mail release notes, I don't think we talked about how to differentiate in the notes yaml.

File: .github/workflows/shippable_builds.yml [hunk 2]
```diff
@@ -572,6 +581,21 @@ jobs:
...
+          APPLICATION_LABEL: ${{ steps.pkginfo.outputs.APPLICATION_LABEL }}
+          FULL_VERSION_NAME: ${{ steps.pkginfo.outputs.FULL_VERSION_NAME }}
+          VERSION_CODE: ${{ steps.pkginfo.outputs.VERSION_CODE }}
+          THUNDERBIRD_GITHUB_NOTES: ${{ needs.release_commit.outputs.thunderbird_github_notes }}
+        run: |
+          if [[ "$APP_NAME" == "thunderbird" ]]; then
+            echo "<h2>${APPLICATION_LABEL} ${FULL_VERSION_NAME} GitHub Draft Release Notes (${VERSION_CODE})</h2><pre>" | tee $GITHUB_STEP_SUMMARY
+            echo "${THUNDERBIRD_GITHUB_NOTES}" | sed -z 's/\n\n/\n/g' | tee -a $GITHUB_STEP_SUMMARY
+            echo "</pre>" | tee -a $GITHUB_STEP_SUMMARY
+          fi
```
Comment (kewisch) [inline .github/workflows/shippable_builds.yml:597]: For the step summary, let's rather show them early. We can do this together with the short form notes.

At this point in the process, we instead need to add them to the publish_gh step below instead so they get added to the github release.

File: scripts/render-notes.py [hunk 3]
```diff
@@ -77,25 +94,35 @@ def render_notes(
...
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument(
-        "--print",
-        "-p",
+        "--print-only",
+        "-po",
         action="store_true",
         help="Only print the processed release notes",
     )
```
Comment (kewisch) [inline scripts/render-notes.py:118]: This was mainly for local debugging, I think one of the arguments is sufficient.

File: scripts/render-notes.py [hunk 4]
```diff
@@ -77,25 +94,35 @@ def render_notes(
...
+    parser.add_argument(
+        "--print",
+        "-p",
+        nargs="+",
+        default=[],
```
Comment (kewisch) [inline scripts/render-notes.py:123]: Adding `choices` here would be good if we're using the identifiers.

File: .github/workflows/shippable_builds.yml (hunk 2 above)
Comment (kewisch) [reply .github/workflows/shippable_builds.yml:597]: What you'll need to do is extend the Get release sha step to output `app_release_notes` based on the appname, and then use this in the publish_gh step.

**Review by kewisch — CHANGES_REQUESTED** [review]
Thanks for handing this in. I have a few comments that will need changes. Get in touch before you get started though, depending on how far along we are with the beta release I might have to work on this beforehand.

File: .github/workflows/shippable_builds.yml (hunk 2 above)
Comment (coreycb) [reply .github/workflows/shippable_builds.yml:597]: Ok. I wasn't sure if we were ready to publish the notes as they are rendered now, or if you wanted to just print them out for now and fine-tune manually.

**Comment by kewisch** [comment]
Ported this to #8380

**Pull request closed without merging**